plugins/dimension-dateTime/src/main/scala/com/stratio/sparkta/plugin/dimension/datetime/DateTimeDimension.scala

```diff
@@ -22,49 +22,61 @@ import java.util.Date
 import akka.event.slf4j.SLF4JLogging
 import org.joda.time.DateTime
 
-import DateTimeDimension._
+import com.stratio.sparkta.plugin.dimension.datetime.DateTimeDimension._
 import com.stratio.sparkta.sdk._
 
 case class DateTimeDimension(props: Map[String, JSerializable]) extends Bucketer with JSerializable with SLF4JLogging {
 
   def this() {
-    this(Map((GRANULARITY_PROPERTY_NAME, DEFAULT_GRANULARITY)))
+    this(Map())
   }
 
-  override val properties: Map[String, JSerializable] = props
+  override val defaultTypeOperation = TypeOp.Timestamp
 
-  override val bucketTypes: Seq[BucketType] = Seq(timestamp, seconds, minutes, hours, days, months, years)
+  override val properties: Map[String, JSerializable] = props ++ {
+    if (!props.contains(GranularityPropertyName)) Map(GranularityPropertyName -> DefaultGranularity) else Map()
+  }
+
+  override val bucketTypes: Map[String, BucketType] = Map(
+    timestamp.id -> timestamp,
+    SecondName -> getPrecision(SecondName, getTypeOperation(SecondName)),
+    MinuteName -> getPrecision(MinuteName, getTypeOperation(MinuteName)),
+    HourName -> getPrecision(HourName, getTypeOperation(HourName)),
+    DayName -> getPrecision(DayName, getTypeOperation(DayName)),
+    MonthName -> getPrecision(MonthName, getTypeOperation(MonthName)),
+    YearName -> getPrecision(YearName, getTypeOperation(YearName)))
 
   @throws(classOf[ClassCastException])
   override def bucket(value: JSerializable): Map[BucketType, JSerializable] =
     try {
       bucketTypes.map(bucketType =>
-        bucketType -> DateTimeDimension.bucket(value.asInstanceOf[Date], bucketType, properties)
-      ).toMap
+        bucketType._2 -> DateTimeDimension.bucket(value.asInstanceOf[Date], bucketType._2, properties))
```

Comment:
We should start avoiding this kind of accesors (._2) and start using pattern matching when possible